src/CarbonAware.CLI/test/unitTests/Commands/EmissionsCommandTests.cs

```diff
@@ -0,0 +1,167 @@
+        var expectedLocations = new List<string>() { longAliasLocation, shortAliasLocation };
+        IEnumerable<string> actualLocations = Array.Empty<string>();
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .Callback((CarbonAwareParameters _parameters) => {
+                actualLocations = _parameters.MultipleLocations.Select(l => l.DisplayName);
+             });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
+
+        // Assert
+        _mockCarbonAwareAggregator.Verify(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()), Times.Once);
+        CollectionAssert.AreEquivalent(expectedLocations, actualLocations);
+    }
+
+    [TestCase("--start-time", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run StartTimeOption: long alias")]
+    [TestCase("-s", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run StartTimeOption: short alias")]
+    public async Task Run_CallsAggregatorWithStartTimeOptions(string alias, string optionValue)
+    {
+        // Arrange
+        var emissionsCommand = new EmissionsCommand();
+        var invocationContext = SetupInvocationContext(emissionsCommand, $"emissions {alias} {optionValue}");
+        var expectedStartTime = DateTimeOffset.Parse(optionValue);
+        DateTimeOffset actualStartTime = default;
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .Callback((CarbonAwareParameters _parameters) => {
+                actualStartTime = _parameters.Start;
+            });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
+
+        // Assert
+        _mockCarbonAwareAggregator.Verify(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()), Times.Once);
+        Assert.AreEqual(expectedStartTime, actualStartTime);
+    }
+
+    [TestCase("--end-time", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run EndTimeOption: long alias")]
+    [TestCase("-e", "2022-01-02T03:04:05Z", TestName = "EmissionsCommandTests.Run EndTimeOption: short alias")]
+    public async Task Run_CallsAggregatorWithEndTimeOptions(string alias, string optionValue)
+    {
+        // Arrange
+        var emissionsCommand = new EmissionsCommand();
+        var invocationContext = SetupInvocationContext(emissionsCommand, $"emissions {alias} {optionValue}");
+        var expectedEndTime = DateTimeOffset.Parse(optionValue);
+        DateTimeOffset actualEndTime = default;
+
+        _mockCarbonAwareAggregator.Setup(agg => agg.GetEmissionsDataAsync(It.IsAny<CarbonAwareParameters>()))
+            .Callback((CarbonAwareParameters _parameters) => {
+                actualEndTime = _parameters.End;
+            });
+
+        // Act
+        await emissionsCommand.Run(invocationContext);
```

Comment:
IMO, we should not test the 'required' flag as part of the unit test. In doing so, we are actually testing that the System.Commandline framework handles the 'required' flag appropriately. In our case, we need to ensure that our client (CLI) calls the aggregator with the appropriate parameters. It is the same in the Controller unit tests, where we do not test the required parameters. This should be handled as part of Integration tests.

Reply:
> IMO, we should not test the 'required' flag as part of the unit test. In doing so, we are actually testing that the System.Commandline framework handles the 'required' flag appropriately. In our case, we need to ensure that our client (CLI) calls the aggregator with the appropriate parameters. It is the same in the Controller unit tests, where we do not test the required parameters. This should be handled as part of Integration tests.

It is not just the required flags. It should be testing all the flags (like `-b` and  `-a` and the expanded versions of those, like some of the utests are doing now.), so basically what the CLI command is doing. The integration test is more high level, since it test functionality as such (like testing the output).
Right now to me this utest is missing the fundamentals of testing what the command(s) should be doing. It is confusing to see that passing some commands without a required flag for instance works. But if I do this and read the code of how a command is implemented, I would find required options.

Reply:
Had an offline conversation with @juzuluag and I got his point. Currently, the unit tests do not depict an accurate picture what options are required for the command to run. 
I tried modifying the unit tests to invoke the Command rather than directly calling the Run() method. However, I ran into an issue where passing a mocked Aggregator was not possible with the current DI approach. Ideally, we should do DI using the constructor like we do elsewhere in the application. This would help create cleaner mocks in the unit tests.
Juan and I agreed that we keep the current PR as is and address this as part of the next PR to keep things moving.